subprojects/persistent-cache/src/main/java/org/gradle/cache/internal/GradleVersionProvider.java

```diff
@@ -0,0 +1,25 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.cache.internal;
+
+import org.gradle.util.GradleVersion;
+
+import java.util.SortedSet;
+
+public interface GradleVersionProvider {
```

Comment:
I'm not too fond of this name. Maybe something like `RecentlyUsedGradleVersions`?

Reply:
I wasn't too fond of it either. I'll rename it to `RecentlyUsedGradleVersions`. Thanks!